### PR TITLE
feat: allow to style BPMN Elements

### LINKED ISSUE #13: [FEAT] Allow to style BPMN Elements
`bpmn-visualization` already provides a CSS API but it requires the CSS styles are defined in the html page. Is this convenient for the R integration? the R package may not let R user to modify the underlying html page.

We may need `bpmn-visualization` to provide a new API that let style specific BPMN elements: https://github.com/process-analytics/bpmn-visualization-js/issues/1030. This feature is available as of bpmn-visualization@0.33.0 which is bundled in bpmnVisualizationR 0.4.0.

### PR DESCRIPTION
**Pull request by csouchet**
This pull request includes changes to several files in the `inst/htmlwidgets` and `R` directories. Here is a description of the changes made in some files:

1. **R/funs.R**:
   - Added a new function `create_element_style` which creates the correct style structure for BPMN elements.
   - Added a new function `create_edge_style` which creates the correct style structure for BPMN edges.
   - Added a new function `create_shape_style` to create the correct style structure for BPMN shapes.
   - Added a new `create_gradient_fill' function to create a gradient fill style for BPMN shapes.
   - Modified the `create_stroke`, `create_font` and `create_fill` functions to accept additional style properties.
   - Updated the `build_bpmnContent` function to accept a `bpmnElementStyles` parameter, which is a list of existing elements with their style to apply.

2. **R/bpmnVisualizationR.R**:
   - Updated the `display` function to accept a `bpmnElementStyles` parameter which allows to specify styles for BPMN elements.
   - An example of how to use `bpmnElementStyles` to define styles for BPMN elements has been added to the function documentation.

### To build the documentation
- https://cran.r-project.org/web/packages/roxygen2/vignettes/reuse.html
- https://process-analytics.github.io/bpmn-visualization-js/api/interfaces/StyleRegistry.html#updateStyle
- https://process-analytics.github.io/bpmn-visualization-js/api/types/Fill.html
- https://process-analytics.github.io/bpmn-visualization-js/api/types/Stroke.html
- https://process-analytics.github.io/bpmn-visualization-js/api/types/Font.html
- https://process-analytics.github.io/bpmn-visualization-js/api/types/StyleWithOpacity.html

Closes #13

### CODE REVIEW

File: inst/htmlwidgets/bpmnVisualizationR.js [hunk 1]
```diff
@@ -50,19 +50,25 @@ HTMLWidgets.widget({
...
+                    }
+                }
+
                 // Add overlays
                 overlays?.map(({elementId, ...rest}) => {
                     const overlayConfig = {...rest};
-                    
+
                     if(enableDefaultOverlayStyle && !(overlayConfig.style && overlayConfig.position)) {
                         const elementsByIds = bpmnVisualization.bpmnElementsRegistry.getElementsByIds(elementId);
```
[the remark this answers is not in the document]
Comment (tbouffard) [reply inst/htmlwidgets/bpmnVisualizationR.js:73]: If we agree on this, I can create a dedicated PR to manage the changes

**Review by tbouffard — CHANGES_REQUESTED** [review]
✔️ Implementation
✔️ Great documentation (including documentation section reuse) 
✔️ A lot of new tests
❌ README
- missing API usage in the README of the repository to be consistent with all existing features that are referenced there
- Consider updating the illustration image: use script of  #72 and #45 (we should put the script in the doc folder instead of keeping it in comments of Pull Requests and probably rename this folder into docs for consistency with other repositories)

--------

⚠️ **Side note about the PR description**: IHMO, there is no need for such a PR description when I am a reviewer. I think it was written by a ChatBot as it is very wordy and doesn't provide the information I expect to find to guide my review and write the merge commit.
- It is mainly a list of added/updated functions and the code and R documentation should be enough to provide the same information. If not, this means that the code and/or the documentation are uncleared
- it only states what is done where I expect to find explanations about why it is done. The code should be clear enough about the what (and if required, add code comments). I would prefer to have a clear text about the "doc section" reuse that is currently hidden in the list of links of the "to build the documentation" paragraph.

I've recently realized that this is becoming the norm in the latest PRs you've opened, and I suggest that it be adapted to help my review. Let's discuss it in person to find a solution that works for both of us.

File: inst/htmlwidgets/bpmnVisualizationR.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (csouchet) [reply inst/htmlwidgets/bpmnVisualizationR.js:73]: It's ok for me :slightly\_smiling\_face:

File: R/funs.R [hunk 2]
```diff
@@ -269,7 +544,7 @@ build_bpmnContent <- function(
     bpmnContent = bpmnContent,
     enableDefaultOverlayStyle = enableDefaultOverlayStyle
   )
-
+
```
[the remark this answers is not in the document]
Comment (csouchet) [reply R/funs.R:547]:
```suggestion

```

**Review by tbouffard — APPROVED** [review]
LGTM
ℹ️ the image in the Readme could still be updated

File: inst/htmlwidgets/bpmnVisualizationR.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tbouffard) [reply inst/htmlwidgets/bpmnVisualizationR.js:73]: Done in #248

File: doc/example_email_voting_with_overlays.svg
[the remark this answers is not in the document]
Comment (csouchet) [reply doc/example_email_voting_with_overlays.svg:1]: I didn't use any script. I just edit the svg image.

[the remark this answers is not in the document]
Comment (csouchet) [reply doc/example_email_voting_with_overlays.svg:1]: I didn't understand which activity that you mention. The label is centered:

[the remark this answers is not in the document]
Comment (tbouffard) [reply doc/example_email_voting_with_overlays.svg:1]: Here is how it is rendered in my browser
image

We however needs the script that is doing this rendering anyway to be able to reproduce it.
I understand the pain about the label repositioning. Be aware of https://github.com/process-analytics/bpmn-visualization-tools/tree/v0.4.0/bpmn-visualization-svg-screenshots for the next time

[the remark this answers is not in the document]
Comment (tbouffard) [reply doc/example_email_voting_with_overlays.svg:1]: As per decision with @csouchet this will be handle with #251 and #252

**Pull request merged**